dev-client/src/components/SiteCard.tsx

```diff
@@ -61,19 +61,20 @@ export const SiteCard = ({
   return (
     <Card
       Header={
-        <Heading variant="h6" color="primary.main">
+        <Heading
+          variant="h6"
+          color="primary.main"
+          maxWidth={isPopover ? '85%' : undefined} // make space for the close button
+        >
           {site.name}
         </Heading>
       }
+      maxWidth={isPopover ? '90%' : undefined} // overall card width
```

Comment:
Issue: Not sure what the right thing is yet, but I don't think this solution works. 
1. In the case where it would have fit on one line, the thumbnail image now goes out of the popover bounds on the bottom
![Screenshot 2024-07-25 at 4 31 16 PM](https://github.com/user-attachments/assets/a24ade8d-1806-4df6-bdfb-eea63daac646)


2. At times, the close button may goes out of the popover bounds on the right
![Screenshot 2024-07-25 at 4 46 53 PM](https://github.com/user-attachments/assets/200c2cc5-22a1-48b8-8718-07ae34b2da51)

Reply:
Which devices were you testing on? I was seeing different behavior on different screen sizes and platforms.

Reply:
I was using the Android Pixel 3a emulator